app/src/androidTest/java/com/github/se/project/ui/StudentEndToEnd.kt

```diff
@@ -59,24 +81,51 @@ class EndToEndTest {
       val onSuccess = invocation.arguments[1] as () -> Unit
       onSuccess() // Simulate a successful update
     }
+    whenever(mockProfileRepository.getProfiles(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+      onSuccess(listOf(mockTutor)) // Simulate a list of profiles with our beloved Ozymandias
+    }
     whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
     whenever(mockLessonRepository.getNewUid()).thenReturn("mockUid")
     whenever(mockLessonRepository.addLesson(any(), any(), any())).thenAnswer { invocation ->
       currentLesson = invocation.arguments[0] as Lesson
       val onSuccess = invocation.arguments[1] as () -> Unit
       onSuccess() // Simulate a successful update
     }
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      currentLesson = invocation.arguments[0] as Lesson
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockLessonRepository.getLessonsForStudent(any(), any(), any())).thenAnswer { invocation
+      ->
+      val onSuccess = invocation.arguments[1] as (List<Lesson>) -> Unit
+      if (currentLesson == null) {
+        onSuccess(emptyList()) // Simulate an empty list of lessons
+      } else {
+        onSuccess(listOf(currentLesson!!))
+      } // Simulate a list of lessons with our current lesson
+    }
+    whenever(mockLessonRepository.deleteLesson(any(), any(), any())).thenAnswer { invocation ->
+      currentLesson = null
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful deletion
+    }
   }
 
   // The test interacts with the UI components to simulate the entire user journey, from logging in
-  // and editing profile information to creating and scheduling a lesson.
+  // and editing profile information, to creating and scheduling a lesson. All 5 types of lessons
+  // displayed to students are tested: requesting a specific tutor, creating an open request, having
+  // a tutor respond to your request, having a confirmed lesson, and having a completed lesson.
   @Test
-  fun signIn_profileEdit_lessonCreation_e2eStudentTest() {
+  fun endToEndStudentTest() {
+    val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+
     // Start the app in test mode
     composeTestRule.setContent {
       PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
     }
-    Thread.sleep(30000)
+    Thread.sleep(50000)
 
```

Comment:
It is good to add a safety measure so that the CI can, it might take a bit longer to pass the test however